ly2video/cli.py

```diff
@@ -952,6 +958,26 @@ def parseOptions():
 
     options = parser.parse_args()
 
+    # TODO: use ConfigArgParse maybe?
+    # Look for a config file and override options if they exist.
+    # this is not a good way to do it, we need to populate options
+    # from file and then parse the command line options which will
+    # override the file options.  For now let's just see if it works.
+    config = ConfigParser()
+
+    if config.read('ly2video.cfg'):
+        section = config['External programs']
+        # We can check to see if the option variable is empty ""
+        if not options.winLilypond:
+            if 'lilypond-windows' in section:
+                options.winLilypond = section['lilypond-windows']
+        if not options.winFfmpeg:
+            if 'ffmpeg-windows' in section:
+                options.winFfmpeg = section['ffmpeg-windows']
+        if not options.winTimidity:
+            if 'timidity-windows' in section:
+                options.winTimidity = section['timidity-windows']
```

Comment:
Similarly here, I think there's probably no reason to limit this to Windows.  Also if an option is already set on the CLI, it shouldn't be overridden by `ly2video.cfg`; the CLI option should take precedence as this is conventional behaviour.

Reply:
> Similarly here, I think there's probably no reason to limit this to Windows.

Agreed, I was just following what was originally there.  I'll get rid of all the win references.

> Also if an option is already set on the CLI, it shouldn't be overridden by `ly2video.cfg`; the CLI option should take precedence as this is conventional behaviour.

I agree, and said in those comments, it was for testing and to start the conversation about ConfigArgParse.  It's an external module that does file and argument parsing.

Otherwise I need to think of a clean way to parse the file first and conditionally apply to options.  That's going to be some ugly code.

If this were c I would do it like this:
```
struct { } options;

parse_file(&options);
parse_args(&options);
```

This way the CLI arguments will overwrite the file arguments.  I don't know the 'python' way to do this using configparser and argparse.  Suggestions welcome.

Reply:
Oh, I only just noticed that you already do the check whether the option is set, so that already prevents it being overridden :+1: 

That said, you could shorten it by skipping both `if` conditionals, e.g.

```python
        options.ffmpeg = options.ffmpeg or section.get('ffmpeg')
```

Unfortunately [Python doesn't have an equivalent of the `||=` operator](http://coldattic.info/post/94/), otherwise it could have been shortened to

```python
        options.ffmpeg ||= section.get('ffmpeg')
```